### DISCUSSION TITLE
Prerendering with hydrated islands?

### DISCUSSION THREAD

**Discussion by keehun** [freshframework/fresh#489] [Q&A]
Hi Deno/Fresh community!

Right now, I have a project in SvelteKit using adapter-static with prerender enabled with crawl. I am using API calls in `load()` which allows Svelte's build system to actually pull the API data at build-time to generate static content. This works well. When I see the build SvelteKit's output, I see the API data in the raw HTML. This is exactly what I want. However, the big downfall is that in Svelte, even with the content prerendered, the `load()` is called _**again**_ in the browser.

I want parts of my page to be completely static as if I generated it with a static site generator. I still want to have the modern conveniences of a framework like Svelte in parts of the page where I want to have first-class interactivity and performant javascript code.

This is where I feel like Deno's Fresh will shine. However, I couldn't figure out from reading the documentation that I can prerender content based on live API calls at build time. Is this something that's already possible? I understand how islands work using preact—now I want to find out more about the prerendering capabilities of Fresh (if any).

Thank you!

**Comment by shishkin** [comment]
There is [no SSG in Fresh](https://github.com/denoland/fresh/discussions/686).

**Comment by dnk8n** [comment]
> prerender content based on live API calls at build time.

I could only see that working with Lume. So I decided to see if I could integrate Lume into Fresh.

Here was the idea:
![fresh-lume](https://user-images.githubusercontent.com/3307753/216958787-53fdb6aa-37ae-4948-bbf2-25ae894136ee.svg)

And progress at: https://github.com/dnk8n/rethinkcode/tree/integrate-lume
